File: trace-api/src/api/api_role.py
#!/usr/bin/env python
# encoding: utf-8
# @author: ZengLei


from typing import Any
from fastapi import APIRouter
from ..obj.tobj_role import RoleForm
from ..obj.vobj_role import RoleObj
from ..obj.tobj_role import Perms
from ..obj import Resp, Page
from ..serv.serv_role import Server
from . import try_log

router = APIRouter()
server = Server()


@router.post("/add_role", summary="添加角色", response_model=Resp[Any])
@try_log(perm=Perms.role_edit)
async def add_role(form: RoleForm):
    return await server.add_role(form) 


@router.delete("/delete_role", summary="删除角色", response_model=Resp[Any])
@try_log(perm=Perms.role_edit)
async def delete_role(code: str):
    return await server.delete_role(code)  


@router.post("/update_role", summary="更新角色", response_model=Resp[Any])
@try_log(perm=Perms.role_edit)
async def update_role(form: RoleForm):
    return await server.update_role(form) 


@router.get("/list_role", summary="查询角色列表", response_model=Resp[Page[RoleObj]])
@try_log(perm=[Perms.role_view, Perms.user_view])
async def list_role(name: str = None, page_index: int = 0, page_size: int = 10):
    return await server.list_role(name, page_index, page_size)


@router.get("/get_role", summary="查询角色详情", response_model=Resp[RoleObj])
@try_log(perm=Perms.role_view)
async def get_role(code: str):
    return await server.get_role(code)

File: trace-web/trace/src/api/ApiRole.ts
import { httpPost, httpGet, httpDelete, C_OK as _C_OK } from "./http";

export const C_OK = _C_OK;

export const add_role = async (params: any) => {
    return await httpPost("/trace-api/role/add_role", params);
};

export const delete_role = async (params: any) => {
    return await httpDelete("/trace-api/role/delete_role", params);
};

export const update_role = async (params: any) => {
    return await httpPost("/trace-api/role/update_role", params);
};

export const list_role = async (params: any) => {
    return await httpGet("/trace-api/role/list_role", params);
};

export const get_role = async (params: any) => {
    return await httpGet("/trace-api/role/get_role", params);
};
